libraries/Microsoft.Bot.Builder.AI.TriggerTrees/TriggerTree.cs

```diff
@@ -0,0 +1,359 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.Bot.Builder.AI.TriggerTrees
+{
+    // Each trigger is normalized to disjunctive normal form and then expanded with quantifiers.
```

Comment:
Thanks for this! It was super helpful to understand the other parts of the PR as well